### ISSUE TITLE
Plugin update to version 0.13.46 not possible (ENOTEMPTY)

### ISSUE THREAD

**Issue by cl4udiu5** [ebaauw/homebridge-hue#1113]
<img width="911" alt="Bildschirmfoto 2022-10-01 um 00 23 08" src="https://user-images.githubusercontent.com/74124378/193364645-9bea7150-208d-414d-b07d-3f464aa611ed.png">

#### Log Messages

```
OS | Raspbian GNU/Linux Bullseye (11)
Homebridge | v1.5.0 (running as service)
Node.js Version | v16.17.1
NPM Version | v8.13.2
```

**Comment by cl4udiu5** [comment]
![Uploading Bildschirmfoto 2022-10-01 um 00.52.12.png…]()

**Comment by ebaauw** [comment]
This is an issue with `npm`, not with Homebridge Hue.  Best delete that directory `.homebridge-hue-…` manually.

**Comment by mandrean** [comment]
I'm running a fresh install of the latest Homebridge version (with `node` upgraded to `v16.17.1`):

Getting the same error. Can't install `homebridge-hue@0.13.46`

**Comment by ebaauw** [comment]
Guess again.  That version has had over 3,500 downloads this week.

**Comment by mandrean** [comment]
And what? Let's not be cocky and over-defensive. And I'm guessing most of those are upgrades, not fresh installs.

If you work in software engineering I'm sure you know that some bugs only manifest themselves under very specific conditions.

**Comment by ebaauw** [comment]
If you work in software engineering, you should know better than guessing and collect facts instead:
```
$ sudo npm -g list homebridhe-hue
/usr/lib
└── (empty)

$ sudo npm -g i homebridge-hue@0.13.46

added 110 packages, and audited 111 packages in 14s

51 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```

**Comment by mandrean** [comment]
Not sure what "homebridhe" [sic] has to do with anything I posted.

Probably the worst thing you can do is being rude towards your fans and users of your project. Very childish and immature behavior.

To anyone else reading this: Yeah can still reproduce. Fresh Homebridge image (latest), Raspberry Pi Zero W, installing `0.13.46` results in an error but `0.13.44` works.

**Comment by ebaauw** [comment]
My bad, that was a type.  I am far from infallible, but I had uninstalled Homebridge Hue just before.  To wit:
```
$ sudo npm -g list homebridge-hue
/usr/lib
└── (empty)

$ sudo npm -g i homebridge-hue@0.13.46

added 110 packages, and audited 111 packages in 14s

51 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```
As I said above: this is an issue with `npm`, nothing to do with the Homebridge Hue code.  You're more than welcome to guess otherwise, but please don't waste my time with that.  If you want my help, rather than accusing me of rudeness, you could provide the actual error message and attach the `npm` log file.

**Comment by mandrean** [comment]
It's not a random guess, but an educated guess ([synonym: hypothesis](https://en.wiktionary.org/wiki/educated_guess#Synonyms)). We do a lot of those at software companies.

It's based on:
- A fresh install eliminates any bad state I myself might have induced
- If `0.13.44` works but `0.13.46` doesn't (after repeated retries), then it points towards something with `0.13.46` even if there isn't a bug in the actual business logic or anything like that. But its a starting point for the investigation
- From experience I know bumping dependencies can sometimes cause unexpected issues, often from the implicit bumps of the transitive dependencies further down the tree
- Comparing the commit of `0.13.46` with `0.13.44` we can indeed see that a lot of transitive deps changed between the two versions, further strengthening the hypothesis: [ebaauw/homebridge-hue@`d5fc99589d68db98ff93ed4fb7b7987af825f8b9..98b9d53e165e49bfa17d52722755a4c43b45bc80`#diff-053150b640](https://github.com/ebaauw/homebridge-hue/compare/d5fc99589d68db98ff93ed4fb7b7987af825f8b9..98b9d53e165e49bfa17d52722755a4c43b45bc80#diff-053150b640a7ce75eff69d1a22cae7f0f94ad64ce9a855db544dda0929316519)
- My `npm` complains something diffuse about the `googleapis` package (which I noticed is a part of the transitive deps changeset as well)

I'll reproduce again and see what the logs said exactly. Another hypothesis I had was that perhaps some of the newer deps require a newer version of `node`, but I upgraded to `v16.17.1` with similar results.
